### PR TITLE
Set deleted_by field when deleting XForms asynchronously

### LINKED ISSUE #1797: Track users who initiate asynchronous xform deletions
### Environmental Information

- Operating System: All
- Onadata version: Master
- Python version: All

### Feature request

Track users who initiate asynchronous xform deletions.

### PR DESCRIPTION
**Pull request by DavisRayM**
### Changes / Features implemented

- Track users who initiate asynchronous deletion of XForms

### Steps taken to verify this change does what is intended

- Updated tests

### Side effects of implementing this change

- None

Closes #1797

### CODE REVIEW

File: onadata/apps/api/viewsets/xform_viewset.py [hunk 1]
```diff
@@ -710,8 +710,11 @@ def delete_async(self, request, *args, **kwargs):
         if request.method == 'DELETE':
             xform = self.get_object()
             resp = {
-                u'job_uuid': tasks.delete_xform_async.delay(xform.pk).task_id,
-                u'time_async_triggered': datetime.now()}
+                u'job_uuid': tasks.delete_xform_async.delay(
+                    xform.pk,
+                    request.user.id).task_id,
+                u'time_async_triggered': datetime.now(),
+                u'triggered_by': request.user.username}
```
Comment (ivermac) [inline onadata/apps/api/viewsets/xform_viewset.py]: Do we need `triggered_by` since we'll be saving the user who triggered the delete?

Comment (DavisRayM) [reply onadata/apps/api/viewsets/xform_viewset.py]: Mostly kept it they're for information purposes (if that makes sense) wanted to be extra verbose. On the first async task status just in case.

TL;DR: We don't really need it

Comment (ivermac) [reply onadata/apps/api/viewsets/xform_viewset.py]: Cool, let's remove it for now and may be add it when we need it

Comment (DavisRayM) [reply onadata/apps/api/viewsets/xform_viewset.py]: I've removed the `triggered_by` field in the latest commits. Please have a look @ivermac
